File: docs/sdks/typescript/smart-cities/examples/quickstart.ts
/**
 * Smart Cities SDK - Quickstart Example
 *
 * This example shows how to:
 * 1. Create a client with API key authentication
 * 2. Create a new city
 * 3. Get real-time city metrics
 */

import { createClient } from '@lydian/smart-cities-sdk';

// Create client with API key authentication
const client = createClient({
  auth: { apiKey: process.env.LYDIAN_API_KEY! }
});

// Create a new smart city
const city = await client.createCity({
  name: 'İstanbul Smart City',
  coordinates: { latitude: 41.0082, longitude: 28.9784 },
  population: 15_840_900,
  timezone: 'Europe/Istanbul'
});

// Get real-time city metrics
const metrics = await client.getCityMetrics(city.cityId);
console.log('Traffic congestion:', metrics.traffic.congestionLevel);
console.log('Energy consumption:', metrics.energy.totalConsumption, 'kWh');
console.log('Air quality index:', metrics.air.aqi);

File: docs/sdks/python/lydian-smart-cities/examples/quickstart.py
"""
Smart Cities SDK - Quickstart Example

This example shows how to:
1. Create a client with API key authentication
2. Create a new city
3. Get real-time city metrics
"""

import os
from lydian_smart_cities import SmartCitiesClient

# Create client with API key authentication
client = SmartCitiesClient(api_key=os.getenv("LYDIAN_API_KEY"))

# Create a new smart city
city = client.create_city(
    name="İstanbul Smart City",
    latitude=41.0082,
    longitude=28.9784,
    population=15_840_900,
    timezone="Europe/Istanbul"
)

print(f"City created: {city.cityId}")

# Get real-time city metrics
metrics = client.get_city_metrics(city.cityId)
print(f"Traffic congestion: {metrics.traffic['congestionLevel']}")
print(f"Energy consumption: {metrics.energy['totalConsumption']} kWh")
print(f"Air quality index: {metrics.air['aqi']}")
